From 731572874730e632a03ec73a3283eff4bfc5e4f2 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Fri, 2 Feb 2024 16:36:30 +0100
Subject: [PATCH 0153/1010] net: dovetail: use file binding to detect oob
 socket

sock->sk might be stale while the encpasulating BSD socket is not, so
dereferencing sock->sk is not always safe.

Since the companion kernel must turn on oob capability for the file
struct associated to an oob-enabled BSD socket, we can use this
information to figure out whether the network representation of the
socket is oob capable as well.

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 net/socket.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/socket.c b/net/socket.c
index b178a99c8a..131fbd8eda 100644
--- a/net/socket.c
+++ b/net/socket.c
@@ -148,7 +148,7 @@ static void sock_show_fdinfo(struct seq_file *m, struct file *f)
 
 static inline bool sock_oob_capable(struct socket *sock)
 {
-	return sock->sk && sock->sk->oob_data;
+	return sock->file && sock->file->oob_data;
 }
 
 int __weak sock_oob_attach(struct socket *sock)
-- 
2.30.2

